WolframLanguageForJupyter/Resources/KernelForWolframLanguageForJupyter.wl

```diff
@@ -279,47 +300,145 @@ If[FailureQ[ioPubSocket] || FailureQ[controlSocket] || FailureQ[inputSocket] ||
 	Quit[];
 ];
 
-toOutText[output_] := 
-	StringJoin[
-		"<pre style=\"",
-		StringJoin[{"&#",ToString[#1], ";"} & /@ ToCharacterCode["font-family: \"Courier New\",Courier,monospace;"]], 
-		"\">",
-		StringJoin[{"&#", ToString[#1], ";"} & /@ ToCharacterCode[ToString[output]]],
-		"</pre>"
-	];
+executeRequestHandler[srm_, frameAssoc_, executionCount_Integer] :=
+	Module[
+		{
+			(* messaging *)
+			replyMsgType, replyContent,
+			ioPubReplyContent, ioPubReplyFrame,
+			(* evaluation *)
+			$jupResEval, $res, $msgs,
+			toOut, errorMessage
+		},
+		replyMsgType = "execute_reply";
+		replyContent = ExportString[Association["status" -> "ok", "execution_count" -> executionCount, "user_expressions" -> {}], "JSON", "Compact" -> True];
+
+		(* print function is global under jupyterEvaluationLoop's block *)
+		printFunction = (sendFrame[
+			ioPubSocket,
+			getFrameAssoc[
+					srm,
+					"stream",
+					ExportString[
+						Association[
+								"name" -> "stdout",
+								"text" -> #1
+						], "JSON", "Compact" -> True
+					]
+					,
+					False
+			]["replyMsg"]
+		]&);
+
+		$jupResEval = ToExpression[frameAssoc["content"]["code"], InputForm, Uninteract];
+		$res = $jupResEval["res"];
+		$msgs = $jupResEval["msgs"];
+		
+		printFunction = Function[#;];
+
+		If[FailureQ[$jupResEval],
+			$res = $Failed;
+			$msgs = jupEval[ToExpression[frameAssoc["content"]["code"], InputForm]]["msgs"];
+		];
 
-toOutImage[output_] := 
-	StringJoin[
-		"<img alt=\"Output\" src=\"data:image/png;base64,",
-		BaseEncode[
-			ExportByteArray[
-				If[Head[output] === Manipulate, output, Rasterize[output]],
-				"PNG"
-			]
-		],
-		"\">"
+		(* format output as purely text, image, or cloud interface *)
+		If[TrueQ[InteractQ[ToExpression[frameAssoc["content"]["code"], InputForm, Hold]]] && $CloudConnected,
```

Comment:
Do we want to check $CloudConnected here? I don't see anything needing it in the following code. And it will take an extra second for it to load cloud-related code.